NoshonNetworks/landver
# feat: add map picker for land registration

- Add `MapPicker` component for visual coordinate selection  
- Integrate map picker into `RegisterLandModal`  
- Replace manual coordinate input with interactive map  
- Add visual marker for selected location  
- Display selected coordinates below map  

## Detailed Information

### Title  
**Add map picker for land registration coordinates**  

### Description  

#### Changes Made  
- Added new `MapPicker` component for visual coordinate selection  
- Integrated interactive map into `RegisterLandModal`  
- Replaced manual coordinate input fields with map interface  
- Added visual marker to show selected location  
- Added coordinate display below map  

#### Features  
- Users can now click on a map to select land coordinates  
- Real-time display of selected latitude and longitude  
- Visual marker shows the exact selected location  
- Map is fully interactive with zoom and pan controls  
- Maintains all existing form validation and submission logic  

#### Technical Details  
- Uses `react-leaflet` for map implementation  
- Dynamically loaded to avoid SSR issues  
- Properly handles marker icon assets  
- Maintains TypeScript type safety  
- Preserves existing form functionality  

### Testing  

To test the changes:  
1. Navigate to the land registration form.  
2. Click on the map to select a location.  
3. Verify the coordinates update.  
4. Complete other form fields.  
5. Submit the form with the selected location.  

---

## Related Issues  
Closes #240  

---

## Type of Change  

- [x] 🐛 Bug fix or ⚙️ Enhancement  
- [x] ✨ New feature or chore (change with no new features or fixes)  

---

## Checklist (select as many as applicable)  

- [x] This pull request is ready to be merged and reviewed.
